### PR TITLE
adds a marine flaregun and makes it fit in flare pouches

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
makes a marine flaregun, makes flareguns tiny, makes them fit in flare pouches (only one though, makes some sense and makes flare pouch a more viable option to give you a reason to pick it over grenade pouch)
flare guns have miniscope level range of zoom now, not as good as binoculars but you can fire flares.
Flaregun flares now stop where you click.

## Why It's Good For The Game
Adds another interesting option to the marine arsenal of weaponry and gives a pouch a little more utility than it used to.

## Changelog
:cl:
add: Added new marine flaregun, it can look out to miniscope distance. Fits in your flare pouches too.
qol: flaregun fired flares now stop where you click.
balance: flareguns now have miniscope level range of zoom on them, when you zoom on them.
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]
Cute lil gun

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -841,6 +841,13 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	icon_state = "t37"
 	attach_icon = "t37_a"
 
+/obj/item/attachable/scope/mini/flaregun
+	name = "long range ironsights"
+	desc = "An unremovable set of long range ironsights for a flaregun."
```
Comment (TiviPlus) [inline code/modules/projectiles/gun_attachables.dm:846]: /obj/item/attachable/scope/unremovable ?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/pouch.dm [hunk 2]
```diff
@@ -545,13 +545,17 @@
...
 /obj/item/storage/pouch/flare
 	name = "flare pouch"
-	desc = "A pouch designed to hold flares. Refillable with a M94 flare pack."
+	desc = "A pouch designed to hold flares and a single flaregun. Refillable with a M94 flare pack."
 	max_w_class = 2
 	storage_slots = 7
 	draw_mode = 1
 	icon_state = "flare"
+
+	storage_type_limits = list(/obj/item/weapon/gun/flare = 1)
```
Comment (TiviPlus) [inline code/game/objects/items/storage/pouch.dm:554]:
```suggestion
	storage_type_limits = list(/obj/item/weapon/gun/flare = 1)
```

File: code/modules/projectiles/guns/misc.dm [hunk 3]
```diff
@@ -5,16 +5,19 @@
...
 	ammo = /datum/ammo/flare
+	w_class = WEIGHT_CLASS_TINY
 	var/num_flares = 1
 	var/max_flares = 1
 	flags_gun_features = GUN_UNUSUAL_DESIGN
 	gun_skill_category = GUN_SKILL_PISTOLS
-	fire_delay = 9
+	fire_delay = 0.5 SECONDS
+
+	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/flaregun)
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/misc.dm:20]:
```suggestion
	fire_delay = 0.5 SECONDS
	starting_attachment_types = list(/obj/item/attachable/scope/unremovable/flaregun)
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**